From 05b6456305e6cd0c7eb79f7bbb3bd31357c291b9 Mon Sep 17 00:00:00 2001
From: Liming Wang <liming.wang@windriver.com>
Date: Thu, 27 May 2010 19:22:23 -0700
Subject: [PATCH 14/15] sched: fix errors for api change

There are two api change when porting deadline patches:

 - In upstream ea87bb7853168434f4a82426dd1ea8421f9e604d,
 enqueue_task has been changed: added a parameter. So change
 the call site to avoid building error.

 - In following commit:
 git://gitorious.org/sched_deadline/linux-deadline.git sched-dl
 commit 0017d735092844118bef006696a750a0e4ef6ebd
 select_task_rq has been changed, but in our tree, it doesn't. So
 change it back to fit our tree.

Signed-off-by:: Liming Wang <liming.wang@windriver>
---
 kernel/sched.c    |    2 +-
 kernel/sched_dl.c |    4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/kernel/sched.c b/kernel/sched.c
index ab52e68..970cb66 100644
--- a/kernel/sched.c
+++ b/kernel/sched.c
@@ -4605,7 +4605,7 @@ static void __change_class_task(struct rq *rq, struct task_struct *p,
 	if (running)
 		p->sched_class->set_curr_task(rq);
 	if (on_rq) {
-		enqueue_task(rq, p, 0);
+		enqueue_task(rq, p, 0, false);
 
 		check_class_changed(rq, p, prev_class, oldprio, running);
 	}
diff --git a/kernel/sched_dl.c b/kernel/sched_dl.c
index 8f59aa2..75e0377 100644
--- a/kernel/sched_dl.c
+++ b/kernel/sched_dl.c
@@ -515,7 +515,7 @@ static void check_dl_preempt_curr(struct task_struct *p, struct rq *rq)
 }
 
 static void
-enqueue_task_dl(struct rq *rq, struct task_struct *p, int flags)
+enqueue_task_dl(struct rq *rq, struct task_struct *p, int flags, bool head)
 {
 	struct sched_dl_entity *dl_se = &p->dl;
 
@@ -758,7 +758,7 @@ static void prio_changed_dl(struct rq *rq, struct task_struct *p,
 
 #ifdef CONFIG_SMP
 static int
-select_task_rq_dl(struct rq *rq, struct task_struct *p, int sd_flag, int flags)
+select_task_rq_dl(struct task_struct *p, int sd_flag, int flags)
 {
 	return task_cpu(p);
 }
-- 
1.6.5.2

